llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -32,6 +35,123 @@ cl::opt<bool> TrackRegPressure(
     "aie-pipeliner-track-regpressure",
     cl::desc("Refuse SWP schedules likely to run into register spills"),
     cl::init(true), cl::Hidden);
+cl::opt<bool> HoistZOLAdjust(
```

Comment:
FYI: If we hoist we can have get a better performance (no clue why). If we don't hoist, we can change the LoopsStart expansion to reuse this non-hoisted add. I already have this change, but I saw no direct gains using this approach.

Reply:
I would put this in a code comment. The rationale of reuse in LoopStart expansion is utterly hidden otherwise.